scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala

```diff
@@ -47,135 +44,34 @@ class H2OMOJOModel(override val uid: String) extends H2OMOJOModelBase[H2OMOJOMod
-          if (supportsCalibratedProbabilities()) {
-            udf[BinomialPredictionExtended, Row] { r: Row =>
-              val pred = easyPredictModelWrapper.predictBinomial(RowConverter.toH2ORowData(r))
-              BinomialPredictionExtended(
-                pred.classProbabilities(0),
-                pred.classProbabilities(1),
-                pred.calibratedClassProbabilities(0),
-                pred.calibratedClassProbabilities(1)
-              )
-            }
-          } else {
-            udf[BinomialPrediction, Row] { r: Row =>
-              val pred = easyPredictModelWrapper.predictBinomial(RowConverter.toH2ORowData(r))
-              BinomialPrediction(
-                pred.classProbabilities(0),
-                pred.classProbabilities(1)
-              )
-            }
-          }
-
-        case ModelCategory.Regression => udf[RegressionPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictRegression(RowConverter.toH2ORowData(r))
-          RegressionPrediction(pred.value)
-        }
-        case ModelCategory.Multinomial => udf[MultinomialPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictMultinomial(RowConverter.toH2ORowData(r))
-          MultinomialPrediction(pred.classProbabilities)
-        }
-        case ModelCategory.Clustering => udf[ClusteringPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictClustering(RowConverter.toH2ORowData(r))
-          ClusteringPrediction(pred.cluster, pred.distances)
-        }
-        case ModelCategory.AutoEncoder => udf[AutoEncoderPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictAutoEncoder(RowConverter.toH2ORowData(r))
-          AutoEncoderPrediction(pred.original, pred.reconstructed)
-        }
-        case ModelCategory.DimReduction => udf[DimReductionPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictDimReduction(RowConverter.toH2ORowData(r))
-          DimReductionPrediction(pred.dimensions)
-        }
-        case ModelCategory.WordEmbedding => udf[WordEmbeddingPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictWord2Vec(RowConverter.toH2ORowData(r))
-          WordEmbeddingPrediction(pred.wordEmbeddings)
-        }
-        case ModelCategory.AnomalyDetection => udf[AnomalyPrediction, Row] { r: Row =>
-          val pred = easyPredictModelWrapper.predictAnomalyDetection(RowConverter.toH2ORowData(r))
-          AnomalyPrediction(pred.score, pred.normalizedScore)
-        }
-        case _ => throw new RuntimeException("Unknown model category " + easyPredictModelWrapper.getModelCategory)
-      }
+  private def canGenerateContributions(model: GenModel): Boolean = {
+    model match {
+      case _: PredictContributionsFactory =>
+        val modelCategory = model.getModelCategory
+        modelCategory == ModelCategory.Regression || modelCategory == ModelCategory.Binomial
```

Comment:
Documentation says: 
>Obtaining SHAP values is possible only from H2OGBM and H2OXGBoost pipeline wrappers and for
regression or binomial problems

Shouldn't we also do a check for `H2OGBM` and `H2OXGBoost`?

Reply:
algorithms who implements `PredictContributionsFactory` can produce SHAP values, currently DRF, GBM & XGBoost are implemementing this interface.

In the documentation I mentioned just H2OXGBoost and H2OGbm as we don't yet have wrapper for H2ODRF, but people can still import mojo from H2O which was created from DRF model and get contributions from there. That is why I would like to keep this check as it is even we can train in SW API just GBM & XGBoost